text/0000-c-string-construction.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: c_string_from_iter
+- Start Date: 2015-02-26
+- RFC PR:
+- Rust Issue:
+
+# Summary
+
+Amend the methods available to construct a `CString` to improve composability
+and follow the conventions emerging elsewhere in the standard library.
+
+# Motivation
+
+The implementation of RFCs [#592][rfc 592] and [#840][rfc 840] has resolved
+most of the issues with the design of `std::ffi::CString`, but some new APIs
+have been created at the same time, needing stabilization. This proposal
+aims at addressing the following issues:
+
+1. The `IntoBytes` trait does not seem wholly justified: it falls short of
+   supporting `IntoIterator`, and has become yet another special-interest trait
+   to care about when designing APIs producing string-like data. 
+
+2. The exposure of `Vec` as an intermediate type for all conversions
+   precludes small-string optimizations, such as an in-place variant
+   implemented in [c_string](https://github.com/mzabaluev/rust-c-str).
+
+3. Stylistic: `Result` as return value type of `new` feels a bit too 'loaded'.
```

Comment:
Hm this is an interesting point. I consider constructors like `File::open` or `TcpStream::connect` to be the "`new` for that type", but I've never considered them less of a constructor in a sense because they return a `Result`.